src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+
+                    }
+                    const radius = parseFloat(catTable[idx].cat[7]);
+
+                    inFields = updateMerge(inFields, 'cattable', {
+                        groupKey: gkey,
+                        fieldKey: 'cattable',
+                        indexClicked: idx,
+                        value: catTable[idx].value
+                    });
+                    inFields = updateMerge(inFields, 'conesize', {
+                        groupKey: gkey,
+                        fieldKey: 'conesize',
+                        max: sizeFactor * radius / 3600
+                    });
+                    break;
+                case FieldGroupCntlr.CHILD_GROUP_CHANGE:
+                    console.log('Child group change called...');
+                    break;
+                default:
+                    break;
+            }
+            return inFields;
+        }
+    };
+};
+
+/**
+ * Return the project elements such as label, value and project is present in each
+ * @param {Object} catmaster master table data
+ * @returns {Array} project option array ready to be used in drop-down component
+ */
+function getProjectOptions(catmaster) {
+    return catmaster.map((o) => {
+        return {label: o.project, value: o.project, proj: o.project};
+    });
+}
+
+/**
+ * Returns the sub project elements from that project
+ * @param {Object} catmaster master table data
+ * @param {String} project the project name
+ * @returns {String} returns subproject string from master table and project name
+ */
+function getSubProjectOptions(catmaster, project) {
+    return catmaster.find((op) => {
+        return op.project === project;
+    }).subproject;
+}
+
+/**
+ * Return the collection of catalog elements based on the project and category selected
+ * @param catmaster master table data
+ * @param project project name
+ * @param subproject name of the category under project
+ * @example cat object example: ["WISE", "AllWISE Database", "AllWISE Source Catalog", "WISE_AllWISE", "wise_allwise_p3as_psd", "334", "747634026", "3600", "<a href='http://irsa.ipac.caltech.edu/Missions/wise.html' target='info'>info</a>", "<a href='http://wise2.ipac.caltech.edu/docs/release/allwise/expsup/sec2_1a.html' target='Column Def'>Column Def</a>", "GatorQuery", "GatorDD"]
+ * @returns {Object} array with ith element is an object which option values and an object with n attribute which ith attribute is corresponding
+ * to ith columns in cols of master table and its value, i.e. ith = 0, 0:'WISE', where cols[0]=projectshort
+ */
+function getCatalogOptions(catmaster, project, subproject) {
+    return catmaster.find((op) => {
+        return op.project === project;
+    }).catalogs.find((c) => {
+        return c.subproject.value === subproject;
+    });
+}
+
+class CatalogDDList extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = Object.assign({...this.state}, {...this.props}, {optList: ''});
+    }
+
```

Comment:
dito Trey.  If you extends component, you should have a 'shouldComponentUpdate'.
If not, write is as a functional component.
function CatalogDDList(props) {
return <div...
}